File: gluecodium/src/test/resources/smoke/skip/output/android/com/example/smoke/SkipTypes.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class SkipTypes extends NativeBase {
    public static final class NotInSwift {
        @NonNull
        public String fooField;
        public NotInSwift(@NonNull final String fooField) {
            this.fooField = fooField;
        }
    }
    public static final class NotInDart {
        @NonNull
        public String fooField;
        public NotInDart(@NonNull final String fooField) {
            this.fooField = fooField;
        }
    }
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected SkipTypes(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
}

File: gluecodium/src/test/resources/smoke/skip/output/android-kotlin/com/example/smoke/SkipTypes.kt
/*

 *
 */

@file:JvmName("SkipTypesExtensions")

package com.example.smoke

import com.example.NativeBase

class SkipTypes : NativeBase {

    class NotInJava {
        @JvmField var fooField: String



        constructor(fooField: String) {
            this.fooField = fooField
        }





    }

    class NotInSwift {
        @JvmField var fooField: String



        constructor(fooField: String) {
            this.fooField = fooField
        }





    }

    class NotInDart {
        @JvmField var fooField: String



        constructor(fooField: String) {
            this.fooField = fooField
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

